step3/src/test/java/rubens/step3/BaconsLawTest.java

```diff
@@ -0,0 +1,60 @@
+package rubens.step3;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import java.io.IOException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+
+public class BaconsLawTest {
+
+	@Spy
+	private BaconsLaw baconsLaw;
+	
+	@Before
+	public void setup() {
+		baconsLaw = Mockito.spy(new BaconsLaw());
+	}
+	
+	@Test
+	public void shouldThrowExceptionWhenActor1IsNotProvided() {
+		assertThrows(IllegalArgumentException.class, () -> baconsLaw.calculateDegrees(null));
+	}
+	
+	@Test
+	public void shouldThrowExceptionWhenActor1EqualsActor2() {
+		String[] args = {"Kevin Bacon,Kevin Bacon"};
+		assertThrows(IllegalArgumentException.class, () -> baconsLaw.calculateDegrees(args));
+	}
+	
+	@Test
+	public void shouldReturnDegree1BetweenKevinBaconAndTomCruise() throws IOException {
+		String[] args = {"Tom Cruise"};
+		assertEquals(new Integer(1), baconsLaw.calculateDegrees(args));
```

Comment:
> # There are 2 degrees of separation between Tom Cruise and Kevin Bacon
> # Ralph Macchio starred with Tom Cruise in The Outsiders
> # John Lithgow starred with Ralph Macchio in Distant Thunders
> # Kevin Bacon starred with John Lithgow in Footloose.

Here I found degree separation = 1 between Tom Cruise and Kevin Bacon
Tom Cruise starred with Chris Penn in All the Right Moves
Chriss Penn starred with Kevin Bacon in Footloose